preprocessing/structs.ts

```diff
@@ -96,121 +161,159 @@ const id_field = (name: string) => {
+      fields.gives,
+    ],
+    additionalDefinitions: `import {Bin} from "mgv_lib/BinLib.sol";
+import {Tick} from "mgv_lib/TickLib.sol";
+import {OfferExtra,OfferUnpackedExtra} from "mgv_lib/OfferExtra.sol";
+
+using OfferExtra for Offer global;
+using OfferUnpackedExtra for OfferUnpacked global;
+`
+  },
 
   /* ## `OfferDetail` */
   //+clear+
   /* `OfferDetail`s hold the maker's address and provision/penalty-related information.
 They have the following fields: */
-  offerDetail: [
-    /* * `maker` is the address that created the offer. It will be called when the offer is executed, and later during the posthook phase. */
-    { name: "maker", bits: 160, type: "address" },
-    /* * <a id="structs.js/gasreq"></a>`gasreq` gas will be provided to `execute`. _24 bits wide_, i.e. around 16M gas. Note that if more room was needed, we could bring it down to 16 bits and have it represent 1k gas increments.
-
-  */
-    fields.gasreq,
-    /*
-       * <a id="structs.js/gasbase"></a>  `offer_gasbase` represents the gas overhead used by processing the offer inside Mangrove + the overhead of initiating an entire order.
-
-    The gas considered 'used' by an offer is the sum of
-    * gas consumed during the call to the offer
-    * `offer_gasbase`
-     
-   (There is an inefficiency here. The overhead could be split into an "offer-local overhead" and a "general overhead". That general overhead gas penalty could be spread between all offers executed during an order, or all failing offers. It would still be possible for a cleaner to execute a failing offer alone and make them pay the entire general gas overhead. For the sake of simplicity we keep only one "offer overhead" value.)
-
-   If an offer fails, `gasprice` gwei is taken from the
-   provision per unit of gas used. `gasprice` should approximate the average gas
-   price at offer creation time.
-
-   `offer_gasbase` is _24 bits wide_ -- note that if more room was needed, we could bring it down to 8 bits and have it represent 1k gas increments.
-
-   `offer_gasbase` is also the name of a local Mangrove
-   parameters. When an offer is created, their current value is copied from Mangrove local configuration.  The maker does not choose it.
-
-   So, when an offer is created, the maker is asked to provision the
-   following amount of wei:
-   ```
-   (gasreq + offer_gasbase) * gasprice
-   ```
-
-    where `offer_gasbase` and `gasprice` are Mangrove's current configuration values (or a higher value for `gasprice` if specified by the maker).
-
-
-    When an offer fails, the following amount is given to the taker as compensation:
-   ```
-   (gasused + offer_gasbase) * gasprice
-   ```
-
-   where `offer_gasbase` and `gasprice` are Mangrove's current configuration values.  The rest is given back to the maker.
+  offerDetail: {
+    fields: [
+      /* * `maker` is the address that created the offer. It will be called when the offer is executed, and later during the posthook phase. */
```

Comment:
```suggestion
      /* * `maker` is the address that created the offer. It will be called when the offer is executed and later during the posthook phase. */
```

I think at some point a "we should should Oxford comma" search-replace went wrong - there are a number of ", and" occurences in non-listing contexts.